### PR TITLE
Fix coverity warning in PDLoadCharacterizations

### PR DESCRIPTION
**Pull request by peterfpeterson**
This adds a constructor so default values for member variables can be set.

**To test:**

Just review the code

Fixes coverity issue 1366006.

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DavidFair** [comment]
I think we can completely remove the constructor with some minor changes:
`hasExtras` isn't used within the code at all so this can easily be removed.

`canColumnNames` is a vector of strings which is used in two places - in `exec` and `readCharInfo`. In `readCharInfo` we use the size of the container to determine how much padding to apply so instead we can just hold this size_t as a member variable. If we use C++11 to initialize on declaration to 0 we no longer need the constructor. 

I've added a git patch file with these changes as I didn't want to start pushing to somebody elses branch (has to be a .txt file because of GitHub sigh): [PDLoadChara_changes.txt](https://github.com/mantidproject/mantid/files/608709/PDLoadChara_changes.txt)

File: Framework/DataHandling/src/PDLoadCharacterizations.cpp
Comment (DavidFair) [inline Framework/DataHandling/src/PDLoadCharacterizations.cpp]: Could we put a comment on where 14 comes from?

**Comment by DavidFair** [comment]
LGTM :shipit: once the tests finish

File: Framework/DataHandling/src/PDLoadCharacterizations.cpp
Comment (quantumsteve) [inline Framework/DataHandling/src/PDLoadCharacterizations.cpp]: `const auto &`?

**Review by quantumsteve — CHANGES_REQUESTED** [review]

**Review by quantumsteve — APPROVED** [review]

**Comment by quantumsteve** [comment]
@peterfpeterson I'll merge this once the build finish.

**Pull request merged**
